pkg/reconciler/buildrun/resources/sources/http.go

```diff
@@ -18,40 +19,31 @@ func AppendHTTPStep(
 	cfg *config.Config,
 	taskSpec *tektonv1beta1.TaskSpec,
 	source buildv1alpha1.BuildSource,
-) {
-	// HTTP is done currently all in a single step, see if there is already one
-	httpStep := findExistingHTTPSourcesStep(taskSpec)
-	if httpStep != nil {
-		httpStep.Container.Args[3] = fmt.Sprintf("%s ; wget \"%s\"", httpStep.Container.Args[3], source.URL)
-	} else {
-		httpStep := tektonv1beta1.Step{
-			Container: corev1.Container{
-				Name:       "sources-http",
-				Image:      cfg.RemoteArtifactsContainerImage,
-				WorkingDir: fmt.Sprintf("$(params.%s-%s)", prefixParamsResultsVolumes, paramSourceRoot),
-				Command: []string{
-					"/bin/sh",
-				},
-				Args: []string{
-					"-e",
-					"-x",
-					"-c",
-					fmt.Sprintf("wget \"%s\"", source.URL),
-				},
-			},
-		}
-
-		// append the git step
-		taskSpec.Steps = append(taskSpec.Steps, httpStep)
+) error {
+	if source.HTTP == nil {
+		return fmt.Errorf("http information for source %s is not specified", source.Name)
 	}
-}
 
-func findExistingHTTPSourcesStep(taskSpec *tektonv1beta1.TaskSpec) *tektonv1beta1.Step {
-	for _, candidateStep := range taskSpec.Steps {
-		if candidateStep.Name == "sources-http" {
-			return &candidateStep
-		}
+	httpStep := tektonv1beta1.Step{
+		Container: corev1.Container{
+			Name:  fmt.Sprintf("source-%s", source.Name),
+			Image: cfg.RemoteArtifactsContainerImage,
+			WorkingDir: filepath.Join(
+				fmt.Sprintf("$(params.%s-%s)", prefixParamsResultsVolumes, paramSourceRoot),
+				source.Destination),
```

Comment:
I think we should tolerate that the destination directory is not yet existing. Will be trivial to handle once we have our own container implemented to perform the HTTP downloads similar to what we have for git. For now, I suggest you always use the source-root as working directory, and then run a script:

```bash
set -euo pipefail

mkdir -p '" + path.join("/workspace/source", source.Destination) +"'
cd '" + path.join("/workspace/source", source.Destination) +"'
wget '" + source.HTTP.URL + "'"
```

Reply:
Agree that we should tolerate that the directory doesn't need to exist in the container beforehand - in fact, we should expect this because the data is going to get downloaded into the source Tekton workspace.